klayout/drc/rule_decks/main.drc

```diff
@@ -14,880 +14,772 @@
-#===========================================================================================================================
+#=======================================================================================================================
+#--------------------------------------------- GF 0.18um MCU DRC RULE DECK ---------------------------------------------
+#=======================================================================================================================
 require 'time'
-require "logger"
+require 'logger'
 require 'etc'
 
 exec_start_time = Time.now
 
-logger = Logger.new(STDOUT)
+logger = Logger.new($stdout)
 
-logger.formatter = proc do |severity, datetime, progname, msg|
-  "#{datetime}: Memory Usage (" + `pmap #{Process.pid} | tail -1`[10,40].strip + ") : #{msg}
+logger.formatter = proc do |_severity, datetime, _progname, msg|
+  "#{datetime}: Memory Usage (" + `pmap #{Process.pid} | tail -1`[10, 40].strip + ") : #{msg}
 "
 end
 
 #================================================
 #----------------- FILE SETUP -------------------
 #================================================
-logger.info("Starting running GF180MCU Klayout DRC runset on %s" % [$input])
-logger.info("Ruby Version for klayout: %s" % [RUBY_VERSION])
+logger.info("Starting running GF180MCU Klayout DRC runset on #{$input}")
+logger.info("Ruby Version for klayout: #{RUBY_VERSION}")
 
 if $input
-    if $topcell
-        source($input, $topcell)
-    else
-        source($input)
-    end
-end
-
-if $table_name
-  table_name = $table_name
-else
-  table_name = "main"
+  if $topcell
+    source($input, $topcell)
+  else
+    source($input)
+  end
 end
 
-
-logger.info("Loading database to memory is complete.")
+logger.info('Loading database to memory is complete.')
 
 if $report
-    logger.info("GF180MCU Klayout DRC runset output at: %s" % [$report])
-    report("DRC Run Report at", $report)
+  logger.info("GF180MCU Klayout DRC runset output at: #{$report}")
+  report('DRC Run Report at', $report)
 else
-    layout_dir = Pathname.new(RBA::CellView::active.filename).parent.realpath
-    report_path = layout_dir.join("gf180_drc.lyrdb").to_s
-    logger.info("GF180MCU Klayout DRC runset output at default location: %s" % report_path)
-    report("DRC Run Report at", report_path)
+  layout_dir = Pathname.new(RBA::CellView.active.filename).parent.realpath
+  report_path = layout_dir.join('gf180_drc.lyrdb').to_s
+  logger.info('GF180MCU Klayout DRC runset output at default location: %s', report_path)
+  report('DRC Run Report at', report_path)
 end
 
 #================================================
 #------------------ SWITCHES --------------------
 #================================================
-logger.info("Evaluate switches.")
+logger.info('Evaluate switches.')
+
+def bool_check?(obj)
+  obj.to_s.downcase == 'true'
+end
 
 # connectivity rules
-if $conn_drc == "true"
+if $conn_drc == 'true'
   CONNECTIVITY_RULES = $conn_drc
```

Comment:
shouldn't you use `bool_check` here too?

Reply:
Done